kdwsdl2cpp/src/converter.cpp

```diff
@@ -130,14 +128,14 @@ class TypeCollector
                     usedComplexTypes.insert(typeName, complexType);
 
                     addDependency(complexType.baseTypeName());
-                    Q_FOREACH (const QName &derivedTypeName, complexType.derivedTypes()) {
+                    for (const QName &derivedTypeName : complexType.derivedTypes()) {
```

Comment:
FYI:
derivedTypes() returns a QList, so this makes a deep copy. qAsConst() should NOT be used since it's a temporary object. So the "proper" solution is

    const auto derivedTypes = complexType.derivedTypes();

However. This is "just" a code generator, I don't expect the performance difference to be significant, so I'm OK with this going in as is. I however suspect that someone will end up having to make the change anyway, due to warnings from clazy...

Reply:
Oh, I didn't knew about this issue with temporary objects in connection with for-range.
This probably affects all occurences of temporary containers here (24 in this commit).

I'll change the lines marked by you .. how about the other 22 ones?

Reply:
If you can change them all, that's even better :)